### PR TITLE
196 add unit tests to slurfilter package

### PR DESCRIPTION
**Pull request by brandrich**

### CODE REVIEW

File: src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public SlurFilter getHighestSlurFilterMatchingText(String text) {
       return null;
     }
 
-    return slurFilters.stream().filter((slur) -> text.matches(slur.getSlurRegex()))
```
Comment (brandrich) [inline src/main/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterService.java:53]: String.matches() only returns true when the _entire_ string matches. Opted to just fix it here vs creating extra leg work with another issue and juggling tests.

File: src/test/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterServiceUnitTests.java [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+
+    when(slurFilterRepository.findAll()).thenReturn(List.of(replaceFilter, anotherReplaceFilter));
+    when(slurFilterRepository.findBySlurActionType(SlurActionType.REPLACE)).thenReturn(List.of(replaceFilter, anotherReplaceFilter));
+
+    String resultText = slurFilterService.censorText("Some mean words in here.");
+
+    verify(slurFilterRepository, times(1)).findAll();
+    verify(slurFilterRepository, times(1)).findBySlurActionType(SlurActionType.REPLACE);
+
+    assertEquals("Some **** *********** in here.", resultText, "Censored text did not match expected");
```
Comment (brandrich) [inline src/test/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterServiceUnitTests.java:328]: I noticed some odd behavior here that I am unsure is intentional. When censoring text the word getting censored ends up being replaced with a number of '*' equal to the length of the slur regex. In this case the word being censored has a length of 5, but after censoring is replaced by 11 '*'.

Comment (jgrim) [reply src/test/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterServiceUnitTests.java:328]: That's odd... I'll poke around and see what I can find later tonight.

**Review by jgrim — APPROVED** [review]

File: src/test/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterServiceUnitTests.java (hunk 2 above)
Comment (jgrim) [reply src/test/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterServiceUnitTests.java:328]: I was able to recreate this issue. I'll create an issue to have it resolved.

Comment (jgrim) [reply src/test/java/com/sublinks/sublinksapi/slurfilter/services/SlurFilterServiceUnitTests.java:328]: See issue [#262](https://github.com/sublinks/sublinks-api/issues/262)

**Pull request merged**